apps/src/templates/manageStudents/ShowSecret.jsx

```diff
@@ -38,9 +43,22 @@ class ShowSecret extends Component {
   };
 
   reset = () => {
-    this.props.resetSecret();
-    this.setState({
-      isShowing: false
+    $.ajax({
+      url: `/v2/students/${this.props.id}/update`,
+      method: 'POST',
+      contentType: 'application/json;charset=UTF-8',
+      data: JSON.stringify({secrets: "reset"}),
+    }).done((data) => {
+      if (this.props.loginType === SectionLoginType.picture) {
+        this.props.setSecretImage(this.props.id, data.secret_picture_path);
+      } else if (this.props.loginType === SectionLoginType.word) {
+        this.props.setSecretWords(this.props.id, data.secret_words);
+      }
+    }).fail((jqXhr, status) => {
```

Comment:
what's jqXhr?

Reply:
I also had to look this up - it's the short name for the object we get back when we make an ajax request: http://api.jquery.com/jquery.ajax/